### DISCUSSION TITLE
В чем заключается смысл ApplicationId?

### DISCUSSION THREAD

**Discussion by HELLRAiSER555** [vknet/vk#1419] [Q&A]
В чем смысл? какой бы я ApplicationId не ставил - от от кэйт мобайл и от айфона и от Вконтакте и даже без ApplicationId и даже заведомо неверный (ApplicationId = "123фвк") - результат в Истории активности вк одинаковый:
Android
Приложение ВКонтакте
Москва · сегодня в 2:24

**Comment by inyutin-maxim** [comment]
> В чем смысл? какой бы я ApplicationId не ставил - от от кэйт мобайл и от айфона и от Вконтакте и даже без ApplicationId и даже заведомо неверный (ApplicationId = "123фвк") - результат в Истории активности вк одинаковый: Android Приложение ВКонтакте Москва · сегодня в 2:24

Давай по порядку каким образом ты авторизуешься?
- [ ] Стандартным способом через логин пароль из либы
- [ ] Через токен?
- [ ] Через байпас?

**Reply by HELLRAiSER555** [reply]
Я авторизируюсь по логину и паролю чтобы получить токен, а потом уже юзаю полученный токен.
Dim services = New ServiceCollection()
                services.AddAudioBypass
                Dim api = New VkApi(services)
                api.Authorize(New ApiAuthParams With {
                .ApplicationId = 2685278,
                .Login = grdUser.CurrentRow.Cells(1).Value,
                .Password = grdUser.CurrentRow.Cells(0).Value,
                .Settings = Settings.All,
                .TwoFactorAuthorization = Function()
                                              Console.WriteLine("Enter Code:")
                                              Return Console.ReadLine()
                                          End Function
})
                lblToken.Text = api.Token
И еще сразу вопросы
1. Что делает AddAudioBypass? Или перефразирую Каково его предназначение окромя  того ничего, что написано в документации? "Расширение для [VkNet](https://github.com/vknet/vk) для обхода ограничения к методам Audio и Messages." Каких ограничений? Если тех ограничений, которые я знаю, то он ничего не делает. К примеру audio.save Error 9 Flood Control.
2. Как загрузить аудио в сообщество (группу) если это вообще сейчас возможно? Раньше был параметр gid или group_id в методе GetUploadServer.
3. Как избежать FloodControl. Я оборачиваю цикл с вконтактовским audio.Save в execute, но это не спасает. Внутри execute все равно возникает FloodControl.
4. Как с этим обстоят дела в VkNet? Для каждой записи вызывать audio.Save? Не хуже ли это?
